dbms/src/Storages/StorageFile.h

```diff
@@ -71,6 +71,9 @@ class StorageFile : public ext::shared_ptr_helper<StorageFile>, public IStorage
     std::string path;
     int table_fd = -1;
 
+    bool path_with_globs = false;
+    std::vector<std::string> matched_paths;
```

Comment:
We should remove `std::string path` and use `matched_paths` only. If user doesn't use globs, than we will single string in this vector.